Sources/Hummingbird/Middleware/MetricsMiddleware.swift

```diff
@@ -58,24 +58,25 @@ public struct MetricsMiddleware<Context: RequestContext>: RouterMiddleware {
             } else {
                 errorType = "500"
             }
-            // need to create dimensions once request has been responded to ensure
-            // we have the correct endpoint path
-            let dimensions: [(String, String)]
-            // Don't record uri in 404 errors, to avoid spamming of metrics
-            if let endpointPath = context.endpointPath {
-                dimensions = [
+            let endpointPath = context.endpointPath ?? "NotFound"
+            // increment requests
+            Counter(
+                label: "hb.requests",
+                dimensions: [
                     ("http.route", endpointPath),
                     ("http.request.method", request.method.rawValue),
-                    ("error.type", errorType),
+                    ("http.response.status_code", errorType),
                 ]
-                Counter(label: "hb.requests", dimensions: dimensions).increment()
-            } else {
-                dimensions = [
+            ).increment()
+            // increment errors
+            Counter(
+                label: "hb.request.errors",
+                dimensions: [
+                    ("http.route", endpointPath),
                     ("http.request.method", request.method.rawValue),
                     ("error.type", errorType),
```

Comment:
Doesn't this need to change too?

Reply:
The dimensions sent to `hb.request.errors` use `error.type`. The dimensions sent to `hb.requests` use dimension `http.response.status_code`